src/blog/dark-mode-with-style-dictionary/index.mdx

```diff
@@ -0,0 +1,1352 @@
+
+```json
+{
+  "info": {
+    "author": "xcode",
+    "version": 1
+  },
+  "images": [
+    {
+      "idiom": "universal",
+      "filename": "img.png"
+    },
+    {
+      "idiom": "universal",
+      "appearances": [
+        {
+          "appearance": "luminosity",
+          "value": "dark"
+        }
+      ],
+      "filename": "img-dark.png"
+    }
+  ]
+}
+```
+
+We can then use the image in SwiftUI or UIKit
+
+```swift
+// SwiftUI
+Image.init("logo")
+
+// UIKit
+UIImage(named: "logo")
+```
+
+For this article I will be focusing on Swift and SwiftUI, but if your app is still on Objective-C you can still use iOS assets and UIKit. You will need to translate some code accordingly. Here are some more resources about supporting dark mode in iOS:
+
+* [https://developer.apple.com/documentation/xcode/supporting_dark_mode_in_your_interface](https://developer.apple.com/documentation/xcode/supporting_dark_mode_in_your_interface)
+* [https://nshipster.com/dark-mode/](https://nshipster.com/dark-mode/)
+
+To summarize, we will build for iOS is an asset catalog with color and image assets that have light and dark mode. Additionally we will create some Swift extensions for Color and Image to make referencing the design tokens a bit easier.
+
+Now that we have an idea of what we need to build for each platform, lets take a look at how we can structure our tokens to support color modes.
+
+
+----
+
+
+## Token structure
+
+Let me start off by saying: **there is no single right or wrong way to support color modes with design tokens**. Here is just what I have found in my experience, but there could be other ways as well. We will first look at how to structure our tokens and then we will get into different ways to implement dark mode because the overall token structure won't change.
+
+### Multi-Tiered system
+
+<TokenStructure />
+
+The most important tip about implementing dark mode with design tokens is to follow a multi-tiered approach like the one in Brad Frost's article, [Creating themeable design systems](https://bradfrost.com/blog/post/creating-themeable-design-systems/), or something similar. Brad Frost call these tiers: brand definitions, high-level application, and component-specific. They are tiers because each one builds on top of each other, layering on meaning and context. The lowest level is the brand level. I have also seen this called 'core' as well. This tier of design tokens don't reference other tokens, they are raw values.
+
+One thing to note here is that these tiers don't necessarily equate to object, file, or naming structure of the tokens, but is more conceptual based on *reference* structure of the tokens. The lowest level of tokens do not reference other tokens. They also do not have any contextual or semantic meaning, they are just *options*. 
```

Comment:
The lowest level of tokens do not reference other tokens. > The lowest level of tokens does not reference other tokens.